packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -0,0 +1,549 @@
+    _addToken(tokenAddress, handlerAddress);
+  }
+
+  function _addToken(address tokenAddress, address handlerAddress) private {
+    // Check that the contract implements the interface
+    IFeeHandlerSeller(handlerAddress);
+
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.active = true;
+    tokenState.handler = handlerAddress;
+
+    activeTokens.add(tokenAddress);
+  }
+
+  /**
+    @notice Allows the owner to activate a specified token.
+    @param tokenAddress The address of the token to be activated.
+  */
+  function activateToken(address tokenAddress) external onlyOwner {
+    _activateToken(tokenAddress);
+  }
+
+  function _activateToken(address tokenAddress) private {
+    activeTokens.add(tokenAddress);
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.active = true;
+  }
+
+  /**
+    @dev Deactivates the specified token by marking it as inactive.
+    @param tokenAddress The address of the token to deactivate.
+  */
+  function deactivateToken(address tokenAddress) external onlyOwner {
+    _deactivateToken(tokenAddress);
+  }
+
+  function _deactivateToken(address tokenAddress) private {
+    activeTokens.remove(tokenAddress);
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.active = false;
+  }
+
+  /**
+    @notice Allows the owner to set a handler contract for a specified token.
+    @param tokenAddress The address of the token to set the handler for.
+    @param handlerAddress The address of the handler contract to be set.
+  */
+  function setHandler(address tokenAddress, address handlerAddress) external onlyOwner {
+    _setHandler(tokenAddress, handlerAddress);
+  }
+
+  function _setHandler(address tokenAddress, address handlerAddress) private {
+    require(handlerAddress != address(0), "Can't set handler to zero, use deactivateToken");
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.handler = handlerAddress;
+  }
+
+  function removeToken(address tokenAddress) external onlyOwner {
+    _removeToken(tokenAddress);
+  }
+
+  function _removeToken(address tokenAddress) private {
+    _deactivateToken(tokenAddress);
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.handler = address(0);
+  }
+
+  function _sell(address tokenAddress) private onlyWhenNotFrozen nonReentrant {
+    IERC20 token = IERC20(tokenAddress);
+
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    require(tokenState.handler != address(0), "Handler has to be set to sell token");
+    FixidityLib.Fraction memory balanceOfTokenToBurn = FixidityLib.newFixed(
+      token.balanceOf(address(this)).sub(tokenState.toDistribute)
+    );
+
+    uint256 balanceToBurn = (burnFraction.multiply(balanceOfTokenToBurn).fromFixed());
+
+    uint256 contractBalance = token.balanceOf(address(this));
+
+    tokenState.toDistribute += (contractBalance.sub(balanceToBurn));
```

Comment:
```suggestion
    tokenState.toDistribute = tokenState.toDistribute.add(contractBalance.sub(balanceToBurn));
```

Reply:
I was doing +=, so the behaviour is the same but it has overflow check with your suggestion.